### PR TITLE
[Feature] Hide Prime Promotions module

### LINKED ISSUE #3374: [Feature Request] Hide Twitch Prime trial banner
I don't see any "Close" button here:

image

But this information is useless for me. And there is annoying color.

### PR DESCRIPTION
**Pull request by whitecrownclown**
@night , 
Fixes #3374
This PR hides the prime banner at the top of the homepage (default: true). Also renamed the autoplay module to homepage and bundled this together.
Waiting on feedback, thanks!

### CODE REVIEW

**Comment by AlexWayfer** [comment]
Probably this should be a part of "Hide Prime Promotions".

**Comment by whitecrownclown** [comment]
@AlexWayfer Thanks for the suggestion, i've moved prime related stuff into their own module.

File: src/modules/hide_prime_promotions/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        settings.on('changed.hideTwitchPrimeBanner', () => this.toggleTwitchPrimeBar());
+        settings.on('changed.hidePrimePromotion', () => this.togglePrimePromotions());
+
+        watcher.on('load.homepage', () => this.toggleTwitchPrimeBar());
+        watcher.on('load', () => this.togglePrimePromotions());
+    }
+
+    toggleTwitchPrimeBar() {
+        $(TWITCH_PRIME_BANNER_SELECTOR).toggleClass(
```
Comment (night) [inline src/modules/hide_prime_promotions/index.js]: we prefer to set classes on body instead of targeting elements directly. due to the nature of react elements can be remounted at any moment, triggering custom classes to be purged from the dom

File: src/modules/hide_prime_promotions/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+const TWITCH_PRIME_BANNER_SELECTOR =
+    '[data-test-selector="test_selector_prime_tracking_button_wrapper"] [data-a-target="blue-bar"]';
+
+class HidePrimePromotionsModule {
+    constructor() {
+        settings.add({
+            id: 'hideTwitchPrimeBanner',
```
Comment (night) [inline src/modules/hide_prime_promotions/index.js]: we should *not* have more settings for this.. one and done. every new setting added is setting bloat

File: src/modules/homepage/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const settings = require('../../settings');
 const watcher = require('../../watcher');
 const twitch = require('../../utils/twitch');
 
-class DisableHomepageAutoplayModule {
+class HomepageModule {
```
Comment (night) [inline src/modules/homepage/index.js]: why are these changes here?

Comment (whitecrownclown) [reply src/modules/homepage/index.js]: Yes, these shouldn't be here.
